[1m============================= test session starts ==============================[0m
platform linux -- Python 3.10.12, pytest-6.2.5, py-1.10.0, pluggy-0.13.0
cachedir: /home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/.pytest_cache
rootdir: /home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing
plugins: launch-testing-1.0.10, ament-lint-0.12.14, ament-copyright-0.12.14, ament-xmllint-0.12.14, launch-testing-ros-0.19.10, ament-pep257-0.12.14, ament-flake8-0.12.14, colcon-core-0.19.0, cov-3.0.0
[1mcollecting ... [0m[1m
collected 1 item                                                               [0m

test/test_integration.py [INFO] [launch]: All log files can be found below /home/phr/.ros/log/2025-08-05-22-38-49-538904-PHR-53256
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [number_publisher-1]: process started with pid [53270]
[INFO] [number_processor-2]: process started with pid [53272]
[number_processor-2] [INFO] [1754404729.655906287] [number_processor]: 数字处理节点已启动，等待接收数字...
[number_publisher-1] [INFO] [1754404729.655931221] [number_publisher]: 数字发布节点已启动，将从1开始发布...
[number_publisher-1] [INFO] [1754404730.650856922] [number_publisher]: 发布: 1
[number_processor-2] [INFO] [1754404730.651139502] [number_processor]: 收到: 1
[number_processor-2] [INFO] [1754404730.651552688] [number_processor]: 发布处理后结果: 3
[number_publisher-1] [INFO] [1754404731.651148110] [number_publisher]: 发布: 2
[number_processor-2] [INFO] [1754404731.651366575] [number_processor]: 收到: 2
[number_processor-2] [INFO] [1754404731.651966169] [number_processor]: 发布处理后结果: 6
[number_publisher-1] [INFO] [1754404732.651166228] [number_publisher]: 发布: 3
[number_processor-2] [INFO] [1754404732.651358573] [number_processor]: 收到: 3
[number_processor-2] [INFO] [1754404732.652029406] [number_processor]: 发布处理后结果: 9
[number_publisher-1] [INFO] [1754404733.651049005] [number_publisher]: 发布: 4
[number_processor-2] [INFO] [1754404733.651412324] [number_processor]: 收到: 4
[number_processor-2] [INFO] [1754404733.651976298] [number_processor]: 发布处理后结果: 12
[number_publisher-1] [INFO] [1754404734.651049339] [number_publisher]: 发布: 5
[number_processor-2] [INFO] [1754404734.651375851] [number_processor]: 收到: 5
[number_processor-2] [INFO] [1754404734.651926765] [number_processor]: 发布处理后结果: 15
[number_publisher-1] [INFO] [1754404735.651087680] [number_publisher]: 发布: 6
[number_processor-2] [INFO] [1754404735.651422503] [number_processor]: 收到: 6
[number_processor-2] [INFO] [1754404735.652028033] [number_processor]: 发布处理后结果: 18
[number_publisher-1] [INFO] [1754404736.651143843] [number_publisher]: 发布: 7
[number_processor-2] [INFO] [1754404736.651377744] [number_processor]: 收到: 7
[number_processor-2] [INFO] [1754404736.651909661] [number_processor]: 发布处理后结果: 21
[number_publisher-1] [INFO] [1754404737.651107408] [number_publisher]: 发布: 8
[number_processor-2] [INFO] [1754404737.651393551] [number_processor]: 收到: 8
[number_processor-2] [INFO] [1754404737.651961088] [number_processor]: 发布处理后结果: 24
[number_publisher-1] [INFO] [1754404738.651048427] [number_publisher]: 发布: 9
[number_processor-2] [INFO] [1754404738.651331008] [number_processor]: 收到: 9
[number_processor-2] [INFO] [1754404738.651884297] [number_processor]: 发布处理后结果: 27
[number_publisher-1] [INFO] [1754404739.651124813] [number_publisher]: 发布: 10
[number_processor-2] [INFO] [1754404739.651345653] [number_processor]: 收到: 10
[number_processor-2] [INFO] [1754404739.651897754] [number_processor]: 发布处理后结果: 30
[number_publisher-1] [INFO] [1754404740.651086041] [number_publisher]: 发布: 11
[number_processor-2] [INFO] [1754404740.651488541] [number_processor]: 收到: 11
[number_processor-2] [INFO] [1754404740.652104757] [number_processor]: 发布处理后结果: 33
[number_publisher-1] [INFO] [1754404741.651072215] [number_publisher]: 发布: 12
[number_processor-2] [INFO] [1754404741.651504398] [number_processor]: 收到: 12
[number_processor-2] [INFO] [1754404741.652141986] [number_processor]: 发布处理后结果: 36
[number_publisher-1] [INFO] [1754404742.651041780] [number_publisher]: 发布: 13
[number_processor-2] [INFO] [1754404742.651300614] [number_processor]: 收到: 13
[number_processor-2] [INFO] [1754404742.651821845] [number_processor]: 发布处理后结果: 39
[WARNING] [launch]: user interrupted with ctrl-c (SIGINT)
[number_processor-2] Traceback (most recent call last):
[number_processor-2]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor", line 33, in <module>
[number_processor-2]     sys.exit(load_entry_point('my-algorithms==0.0.0', 'console_scripts', 'number_processor')())
[number_processor-2]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages/my_algorithms/number_processor_node.py", line 32, in main
[number_processor-2]     rclpy.spin(node)
[number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/__init__.py", line 229, in spin
[number_processor-2]     executor.spin_once()
[number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 751, in spin_once
[number_processor-2]     self._spin_once_impl(timeout_sec)
[number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 740, in _spin_once_impl
[number_processor-2]     handler, entity, node = self.wait_for_ready_callbacks(timeout_sec=timeout_sec)
[number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 723, in wait_for_ready_callbacks
[number_processor-2]     return next(self._cb_iter)
[number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 620, in _wait_for_ready_callbacks
[number_processor-2]     wait_set.wait(timeout_nsec)
[number_processor-2] KeyboardInterrupt
[number_publisher-1] Traceback (most recent call last):
[number_publisher-1]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_publisher", line 33, in <module>
[number_publisher-1]     sys.exit(load_entry_point('my-algorithms==0.0.0', 'console_scripts', 'number_publisher')())
[number_publisher-1]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages/my_algorithms/number_publisher_node.py", line 25, in main
[number_publisher-1]     rclpy.spin(node)
[number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/__init__.py", line 229, in spin
[number_publisher-1]     executor.spin_once()
[number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 751, in spin_once
[number_publisher-1]     self._spin_once_impl(timeout_sec)
[number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 740, in _spin_once_impl
[number_publisher-1]     handler, entity, node = self.wait_for_ready_callbacks(timeout_sec=timeout_sec)
[number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 723, in wait_for_ready_callbacks
[number_publisher-1]     return next(self._cb_iter)
[number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 620, in _wait_for_ready_callbacks
[number_publisher-1]     wait_set.wait(timeout_nsec)
[number_publisher-1] KeyboardInterrupt
[ERROR] [number_publisher-1]: process has died [pid 53270, exit code -2, cmd '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_publisher --ros-args -r __node:=number_publisher'].
[ERROR] [number_processor-2]: process has died [pid 53272, exit code -2, cmd '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor --ros-args -r __node:=number_processor'].
Processes under test stopped before tests completed
Process 'number_publisher-1' exited with -2
##### 'number_publisher-1' output #####
[31mF[0m

=================================== FAILURES ===================================
[31m[1m________________________ launch tests: test_integration ________________________[0m
[1m[31m/usr/lib/python3/dist-packages/_pytest/runner.py[0m:311: in from_call
    result: Optional[TResult] = func()
[1m[31m/usr/lib/python3/dist-packages/_pytest/runner.py[0m:255: in <lambda>
    [94mlambda[39;49;00m: ihook(item=item, **kwds), when=when, reraise=reraise
[1m[31m/usr/lib/python3/dist-packages/pluggy/hooks.py[0m:286: in __call__
    [94mreturn[39;49;00m [96mself[39;49;00m._hookexec([96mself[39;49;00m, [96mself[39;49;00m.get_hookimpls(), kwargs)
[1m[31m/usr/lib/python3/dist-packages/pluggy/manager.py[0m:92: in _hookexec
    [94mreturn[39;49;00m [96mself[39;49;00m._inner_hookexec(hook, methods, kwargs)
[1m[31m/usr/lib/python3/dist-packages/pluggy/manager.py[0m:83: in <lambda>
    [96mself[39;49;00m._inner_hookexec = [94mlambda[39;49;00m hook, methods, kwargs: hook.multicall(
[1m[31m/usr/lib/python3/dist-packages/pluggy/callers.py[0m:208: in _multicall
    [94mreturn[39;49;00m outcome.get_result()
[1m[31m/usr/lib/python3/dist-packages/pluggy/callers.py[0m:80: in get_result
    [94mraise[39;49;00m ex[[94m1[39;49;00m].with_traceback(ex[[94m2[39;49;00m])
[1m[31m/usr/lib/python3/dist-packages/pluggy/callers.py[0m:187: in _multicall
    res = hook_impl.function(*args)
[1m[31m/usr/lib/python3/dist-packages/_pytest/runner.py[0m:170: in pytest_runtest_call
    [94mraise[39;49;00m e
[1m[31m/usr/lib/python3/dist-packages/_pytest/runner.py[0m:162: in pytest_runtest_call
    item.runtest()
[1m[31m/opt/ros/humble/lib/python3.10/site-packages/launch_testing/pytest/hooks.py[0m:112: in runtest
    results_per_run = runner.run()
[1m[31m/opt/ros/humble/lib/python3.10/site-packages/launch_testing/test_runner.py[0m:256: in run
    results[run] = worker.run()
[1m[31m/opt/ros/humble/lib/python3.10/site-packages/launch_testing/test_runner.py[0m:167: in run
    [96mself[39;49;00m._print_process_output_summary(proc_info, proc_output)
[1m[31m/opt/ros/humble/lib/python3.10/site-packages/launch_testing/test_runner.py[0m:209: in _print_process_output_summary
    [96mprint[39;49;00m([33m'[39;49;00m[33m{}[39;49;00m[33m'[39;49;00m.format(io.text.decode([33m'[39;49;00m[33mascii[39;49;00m[33m'[39;49;00m)))
[1m[31mE   UnicodeDecodeError: 'ascii' codec can't decode byte 0xe6 in position 50: ordinal not in range(128)[0m
- generated xml file: /home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/pytest.xml -
=========================== short test summary info ============================
FAILED test/test_integration.py::test_integration - UnicodeDecodeError: 'asci...
[31m============================== [31m[1m1 failed[0m[31m in 14.01s[0m[31m ==============================[0m
